examples/common/convert_c4.py

```diff
@@ -4,105 +4,344 @@
 """C4 streaming dataset conversion scripts."""
 import os
 import platform
+import warnings
 from argparse import ArgumentParser, Namespace
-from typing import Dict, Iterable
+from enum import Enum
+from textwrap import dedent
+from typing import Dict, Iterable, Optional
 
 import datasets as hf_datasets
+import numpy as np
 from streaming import MDSWriter
 from torch.utils.data import DataLoader, IterableDataset, get_worker_info
 from tqdm import tqdm
+from transformers import AutoTokenizer, PreTrainedTokenizerBase
+
+
+class ConcatMode(Enum):
+    NO_CONCAT = 'NO_CONCAT'
+    CONCAT_TEXT = 'CONCAT_TEXT'
+    CONCAT_TOKENS = 'CONCAT_TOKENS'
```

Comment:
Is this needed? why not just let `args.concat_tokens` and `args.concat_text` handle this?

Reply:
The mode is derivable from those. The hope was this made it clearer, but I can remove

Reply:
I think the mode makes it clearer, so will leave it.